lib/rules/no-dupe-characters-character-class.ts

```diff
@@ -179,7 +163,7 @@ function groupingElements(
             }
         } else if (e.type === "CharacterClassRange") {
             const nodeAndCharSet = { node: e, charSet }
-            const range = getMinimumRange(nodeAndCharSet)
+            const range = charSet.ranges[0]
             const key = String.fromCodePoint(range.min, range.max)
```

Comment:
Unfortunately, this doesn't uniquely identify a range. It's almost correct and it was difficult to find an example but here it is:

- `/[A-Z]/i` creates the ranges: `{ min: 65, max: 90 }, { min: 97, max: 122 }`
- `/[a-\uFFFF]/i` creates the ranges: `{ min: 65, max: 90 }, { min: 97, max: 65535 }`

As you can see, `ranges[0]` is the same for both.

The problem is that `ranges[0]` only works when there is no canonicalization (ignore case). Since we cannot assume anything about the canonicalization function, we have to use the whole char set as the key.
```suggestion
            const key = flags.ignoreCase ? charSet.toString() : String.fromCodePoint(charSet.ranges[0].min, charSet.ranges[0].max)
```

Reply:
Thank you for letting me know about this!
I didn't understand the range correctly 😓. I would like to fix this issues.

Reply:
No worries. It also took me quite some time to figure out that this doesn't work.